samples/idTokenFromMetadataServer.js

```diff
@@ -0,0 +1,51 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Uses the Google Cloud metadata server environment to create an identity token
+ * and add it to the HTTP request as part of an Authorization header.
+ *
+ * @param {string} url - The url or target audience to obtain the ID token for.
+ */
+function main(url) {
+  // [START auth_cloud_idtoken_metadata_server]
+  /**
+   * TODO(developer):
+   *  1. Uncomment and replace these variables before running the sample.
+   */
+  // const url = 'http://www.example.com';
+
+  const {GoogleAuth} = require('google-auth-library');
+
+  async function getIdTokenFromMetadataServer() {
+    const googleAuth = new GoogleAuth();
+    const client = await googleAuth.getClient();
+
+    // Get the ID token.
+    // Once you've obtained the ID token, you can use it to make an authenticated call
+    // to the target audience.
+    await client.fetchIdToken(url);
+    console.log('Generated ID token.');
```

Comment:
It would be more helpful to do something with this token rather than to log this message

Reply:
The same for Java and Python: https://github.com/googleapis/google-auth-library-java/blob/main/samples/snippets/src/main/java/IdTokenFromImpersonatedCredentials.java#L83

So, we show how to generate it, user then could modify this part